drivers/flash/flash_shell.c

```diff
@@ -52,22 +43,28 @@ static int cmd_erase(const struct shell *shell, size_t argc, char *argv[])
 	if (argc > 2) {
 		size = strtoul(argv[2], NULL, 16);
 	} else {
-#if defined(CONFIG_SOC_FLASH_NRF)
-		size = NRF_FICR->CODEPAGESIZE;
-#else
-		error(shell, "Missing size.");
-		return -EINVAL;
-#endif
+		struct flash_pages_info info;
+
+		result = flash_get_page_info_by_offs(flash_dev, page_addr,
+						     &info);
+
+		size = info.size;
+
+		if (result != 0) {
+			shell_error(shell, "Could not determine page size, "
+				    "code %d.", result);
+			return result;
```

Comment:
I'm not sure if the convention is to return the actual error code of the underlying call or else to restrict to a set accepted by the shell itself.

Reply:
I am also not sure about that, however this is consistent with the rest of the code, which already return error code like that in case of erase failure for example.

Reply:
Same here but as majority of the flash shell returns it oven value I will do the same (return EINVAL)